### PR TITLE
Fix the proto.yml github workflow file

### PR DESCRIPTION
**Pull request by dwedul-figure**
## Description

1. Adds `if: always()` to each step so that they run even if a previous one failed.
2. Removes the `env_init` job since it was broken (didn't have any `steps`) and `go` isn't needed for these jobs.
3. Runs when `go.mod` is changed because that might indicate a third_party update is needed.
4. Runs when that workflow file is changed so it's easier to tweak and test.
5. Reorders checks to: 1) Lint, 2) Breaking, 3) third party breaking, 4) third party update check. Lint and breaking deal with our own stuff, so I felt like having them first would be nice. Then, since the third party update check takes 30ish seconds (the others take 1ish), I put that last. The order probably doesn't really matter, but my pedantic nature kicked in and decided it wanted a different order.

Related to #843 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by egaxhaj-figure** [comment]
@dwedul-figure check out #843. I address these issues there.

**Review by nullpointer0x00 — DISMISSED** [review]

**Comment by dwedul-figure** [comment]
Blocked until #843 is merged.

**Comment by dwedul-figure** [comment]
These changes have been included in #843.

**Pull request closed without merging**
